### ISSUE TITLE
Upgrade to latest Mojolocious breaks because delay helper was deprecated

### ISSUE THREAD

**Issue by kensanata** [jhthorsen/mojolicious-plugin-cgi#29]
This is what I see in my Toadfarm log (and all my sites are down):
```
[Fri Aug 17 19:31:16 2018] [error] Can't locate object method "delay" via package "Mojolicious::Controller" at /home/alex/perl5/perlbrew/perls/perl-5.26.1/lib/site_perl/5.26.1/Mojolicious/Plugin/CGI.pm line 180.
```

This seems to related to this commit: https://github.com/kraih/mojo/commit/bd2a1e48c7d5ed3df4235fbe674009c386da3e0c

**Comment by kensanata** [comment]
Cheap fix:

```diff
*** /home/alex/perl5/perlbrew/perls/perl-5.26.1/lib/site_perl/5.26.1/Mojolicious/Plugin/CGI.pm~2017-06-12 21:05:08.000000000 +0200
--- /home/alex/perl5/perlbrew/perls/perl-5.26.1/lib/site_perl/5.26.1/Mojolicious/Plugin/CGI.pm2018-08-17 19:53:18.157190521 +0200
***************
*** 30,35 ****
--- 30,45 ----
    $args->{run} = delete $args->{script} if ref $args->{script} eq 'CODE';
    $args->{pids} = $pids;
  
+   # FIXME: got removed in https://github.com/kraih/mojo/commit/bd2a1e48c7d5ed3df4235fbe674009c386da3e0c
+   $app->helper(
+     delay => sub {
+       my $c  = shift;
+       my $tx = $c->render_later->tx;
+       Mojo::IOLoop->delay(@_)
+ 	  ->catch(sub { $c->helpers->reply->exception(pop) and undef $tx })->wait;
+     }
+       );
+   
    $app->helper('cgi.run' => sub { _run($args, @_) }) unless $app->renderer->helpers->{'cgi.run'};
    $app->{'mojolicious_plugin_cgi.tid'}
      ||= Mojo::IOLoop->recurring(CHECK_CHILD_INTERVAL, sub { local ($?, $!); _waitpids($pids); });
```

**Comment by jhthorsen** [comment]
Fixed in 0.39.

Thanks for reporting in 👍
